src/serializer/ResidualHeapSerializer.js

```diff
@@ -1671,6 +1674,15 @@ export class ResidualHeapSerializer {
 
   _serializeAbstractValue(val: AbstractValue): void | BabelNodeExpression {
     invariant(val.kind !== "sentinel member expression", "invariant established by visitor");
+    if (val.kind === "conditional") {
+      let implications = this.conditionalImplications.get(val);
+      invariant(implications !== undefined);
+      if (implications.t && !implications.f) return this.serializeValue(val.args[1]);
+      else if (!implications.t && implications.f) return this.serializeValue(val.args[2]);
+      else if (!implications.t && !implications.f) return voidExpression; // Can this happen?
+      invariant(implications.t && implications.f);
```

Comment:
I would not be very surprised if this case were the common one. It seems to me that your approach is path insensitive, which limits optimization opportunities.